back/models/category_model.py

```diff
@@ -0,0 +1,33 @@
+from sqlalchemy import Column, Integer, String
+from sqlalchemy.orm import validates
+from back.models.base_model import BaseModel
+
+
+class Category(BaseModel):
+    __tablename__ = 'category'
+    name = Column('name', String(length=100), nullable = False)
+    description = Column('description', String(length=255), nullable = True)
+
+    def __init__(self, name: str, description: str) -> None:
+        self.name = name
+        self.description = description
+
+
+@validates('name')
+def validate_name(self, key, name):
+    if not name:
+        raise ValueError("The name can't be empty")
+    if not isinstance(name, str):
+        raise TypeError("The name must be a string")
```

Comment:
inverter a ordem desses 2 if e no if not name, colocar .strip(), para remover os espaços.
if not name.strip()

Reply:
Muito obrigado, Thiago. Resolvido!